numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1839,6 +1839,11 @@ array_empty(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kwds)
     return (PyObject *)ret;
 
 fail:
+    if (PyErr_Occurred() && PyErr_ExceptionMatches(PyExc_TypeError)) {
+        PyErr_SetString(PyExc_TypeError,
+                "data type not understood, "
+                "did you mean to use a tuple for size?");
+    }
```

Comment:
This attempts to match the TypeError set by PyArray_DescConverter where the passed value is not a valid type.

It cannot be moved any lower as the converter is used in many other places and the more specific error message may not be applicable to those cases